utils/python/CIME/SystemTests/system_tests_common.py

```diff
@@ -82,16 +93,43 @@ def coupler_log_indicates_run_complete(self):
         return False
 
     def report(self):
-        newestcpllogfile = self._getlatestcpllog()
+        newestcpllogfile = self._get_latest_cpl_log()
         self._check_for_memleak(newestcpllogfile)
 
+    def _component_compare_move(self, suffix):
+        cmd = os.path.join(self._case.get_value("SCRIPTSROOT"), "Tools",
+                           "component_compare_move.sh")
+        rc, out, err = run_cmd("%s -rundir %s -testcase %s -suffix %s" %
+                               (cmd, self._case.get_value('RUNDIR'), self._case.get_value('CASE'), suffix),
+                               ok_to_fail=True)
+        if rc == 0:
+            append_status(out, sfile="TestStatus.log")
+        else:
+            append_status("Component_compare_test.sh failed out: %s\n\nerr: %s\n"%(out,err)
+                          ,sfile="TestStatus.log")
+
+    def _component_compare_test(self, suffix1, suffix2):
+        cmd = os.path.join(self._case.get_value("SCRIPTSROOT"),"Tools",
+                           "component_compare_test.sh")
+        rc, out, err = run_cmd("%s -rundir %s -testcase %s -testcase_base %s -suffix1 %s -suffix2 %s"
+                               %(cmd, self._case.get_value('RUNDIR'), self._case.get_value('CASE'),
+                                 self._case.get_value('CASEBASEID'), suffix1, suffix2), ok_to_fail=True)
+        if rc == 0:
+            append_status(out, sfile="TestStatus")
+        else:
+            append_status("Component_compare_test.sh failed out: %s\n\nerr: %s\n"%(out,err)
+                          ,sfile="TestStatus.log")
+            return False
+
+        return True
+
     def _get_mem_usage(self, cpllog):
         """
         Examine memory usage as recorded in the cpl log file and look for unexpected
         increases.
         """
         memlist = []
-        meminfo = re.compile(".*model date =\s+(\w+).*memory =\s+(\d+\.?\d+).*highwater")
```

Comment:
The old regex was not correct. You need to use a raw string to express a string with backslashes in it; otherwise the backslashses themselves must be escaped.